charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml

```diff
@@ -22,6 +22,7 @@ spec:
       {{- with .Values.controller.metrics.serviceMonitor.interval }}
       interval: {{ . }}
       {{- end }}
+      scrapeTimeout: {{ .Values.controller.metrics.serviceMonitor.scrapeTimeout }}
```

Comment:
maybe it's better to align to other values, like `interval` ? 🤔  WDYT?
```yaml
      {{- with .Values.controller.metrics.serviceMonitor.interval }}
      interval: {{ . }}
      {{- end }}
```

Reply:
Ah but default value is defined. hmm it doesn't make much difference.

Reply:
I would also preffer this method and unsetting the default in values.yaml

Prometheus-operator has no default value on ServiceMonitors, the default is inherited from the global config of prom-operator.
Ref: https://github.com/prometheus-community/helm-charts/blob/94330ab7439a5ab2701bbe8605097a904223d62c/charts/kube-prometheus-stack/charts/crds/crds/crd-servicemonitors.yaml#L470-L473

So I suggest to define it like this in values:
```yaml
      # -- (string) Prometheus ServiceMonitor scrapeTimeout. If empty, Prometheus uses the global scrape timeout unless it is less than the target's scrape interval value in which the latter is used.
       scrapeTimeout: ~
```

Reply:
@mkilchhofer that's make sense, replaced